src/discussions/learners/data/api.test.jsx

```diff
@@ -0,0 +1,115 @@
+import MockAdapter from 'axios-mock-adapter';
+import { Factory } from 'rosie';
+
+import { getAuthenticatedHttpClient } from '@edx/frontend-platform/auth';
+import { initializeMockApp } from '@edx/frontend-platform/testing';
+
+import { initializeStore } from '../../../store';
+import { executeThunk } from '../../../test-utils';
+import { getCoursesApiUrl, getUserProfileApiUrl } from './api';
+import { fetchLearners, fetchUserPosts } from './thunks';
+
+import './__factories__';
+
+const courseId = 'course-v1:edX+DemoX+Demo_Course';
+const courseId2 = 'course-v1:edX+TestX+Test_Course2';
+const coursesApiUrl = getCoursesApiUrl();
+const userProfileApiUrl = getUserProfileApiUrl();
+
+let axiosMock;
+let store;
+const username = 'abc123';
+const learnerCount = 3;
+
+describe('Learner api test cases', () => {
+  beforeEach(async () => {
+    initializeMockApp({
+      authenticatedUser: {
+        userId: 3,
+        username,
+        administrator: true,
+        roles: [],
+      },
+    });
+    Factory.resetAll();
+    store = initializeStore();
+    axiosMock = new MockAdapter(getAuthenticatedHttpClient());
+  });
+
+  afterEach(() => {
+    axiosMock.reset();
+  });
+
+  async function setupLearnerMockResponse(learnerCourseId, statusCode) {
+    const learnersData = Factory.build('learnersResult', {}, {
+      count: learnerCount,
+      pageSize: 6,
+    });
+    axiosMock.onGet(`${coursesApiUrl}${learnerCourseId}/activity_stats/`)
+      .reply(() => [statusCode, learnersData]);
```

Comment:
Please confirm the can we get the status code from the thunk call without explicitly passing here. e.g
axiosMock.onGet(`${coursesApiUrl}${learnerCourseId}/activity_stats/`).reply(200, {learnersData});